service/src/main/java/org/broadinstitute/listener/relay/transport/RelayedRequestPipeline.java

```diff
@@ -80,12 +80,18 @@ public void registerHttpExecutionPipeline(Scheduler scheduler) {
     listenerConnectionHandler
         .receiveRelayedHttpRequests()
         .publishOn(scheduler)
-        .filter(c -> listenerConnectionHandler.isNotPreflight(c.getRequest()))
```

Comment:
The sequence of negative checks were getting a little confusing, so changed it to a `handle` call.

Reply:
I think the `if-statement-in-handle` is equivalent to a chain of `filter..doOnDiscard` in reactor-core. Just felt easier to read to me.

The added Sam call in `/setCookie` could impact performance… but we do it in Leo too, it's cached, and from the UI the `setCookie` call still felt instantaneous to me. 🤷